broker/submissions/routes.py

```diff
@@ -24,11 +25,18 @@ def generate_spreadsheet(submission_uuid):
 
     message = 'The spreadsheet is being generated.'
     if job_not_created(spreadsheet_job) or job_finished(spreadsheet_job):
```

Comment:
I think this could be refined.
If a spreadsheet generation job has started and failed, we would only have a `createdDate`, so we would go to the `else` branch and not start a new job. We don't have any means of differentiating between a job in progress and a failed job.
In addition, but less important, we generate a new spreadsheet if one has not been created ever, or when one has been created and finished. We could check whether there have been updates to the project and only generate in this case.